File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> result;
        unordered_map<string, vector<string>> storage;
        
        for (auto str : strs){
            string key = str;
            sort(key.begin(), key.end());
            storage[key].push_back(str);
        }
        for (auto lst : storage){
            result.push_back(lst.second);
        }
        return result;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        storage = {}
        n = len(strs)
        out = []
        for i in range(n):
            word = strs[i]
            sort = "".join(sorted(word))
            if sort not in storage:
                storage[sort] = [word]
            else:
                storage[sort].append(word)
        out = []
        for elem in storage:
            out.append(storage[elem])
        return out

